File: app/src/main/java/space/levan/memory/utils/NetUtils.java
package space.levan.memory.utils;

import android.content.Context;
import android.net.ConnectivityManager;

import space.levan.memory.app.App;

/**
 * File description
 *
 * @author WangZhiYao
 * @date 2017/10/17
 */

public class NetUtils {

    public static boolean isNetConnected() {
        ConnectivityManager connectivityManager = (ConnectivityManager) App.getInstance()
                .getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        return connectivityManager.getActiveNetworkInfo() != null;
    }
}

File: app/src/main/java/space/levan/memory/utils/NetUtils.kt
package space.levan.memory.utils

import android.content.Context
import android.net.ConnectivityManager
import space.levan.memory.App

/**
 * File description
 *
 * @author WangZhiYao
 * @date 2018/1/15
 */

object NetUtils {

    fun isNetConnected(): Boolean {
        val cm = App.instance!!.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetwork = cm.activeNetworkInfo
        return activeNetwork != null && activeNetwork.isConnectedOrConnecting
    }
}